kratos.gid/apps/MPM/write/write.tcl

```diff
@@ -90,6 +90,31 @@ proc MPM::write::GetPartsGroups { part_type {what "name"} } {
     return $body_groups
 }
 
+
+proc ::MPM::write::GetUsedElements { {get "Objects"} } {
+    set lista [list ]
+    foreach gNode [MPM::write::GetPartsGroups2] {
+        set elem_name [write::getValueByNode [$gNode selectNodes ".//value\[@n='Element']"] ]
+        set e [Model::getElement $elem_name]
+        if {$get eq "Name"} { set e [$e getName] }
+        lappend lista $e
+    }
+    return $lista
+}
+
+
+proc ::MPM::write::GetPartsGroups2 { {get "Objects"} } {
```

Comment:
I think that this function is the same as -> write::getPartsGroupsId
https://github.com/KratosMultiphysics/GiDInterface/blob/21ef2c92d01b63ee68d2aa793b1fc715d2661eaf/kratos.gid/scripts/Writing/Writing.tcl#L413

As the one in writing is a common code, I think you can refer to it.

Reply:
You are completely right. I've changed it in the new commit. Thank you! :)